client/state/sites/vouchers/reducer.js

```diff
@@ -28,6 +33,28 @@ import { itemsSchema } from './schema';
 export const items = ( state = {}, action ) => {
 	const { siteId } = action;
 	switch ( action.type ) {
+		case SITE_VOUCHERS_ASSIGN:
+			switch ( action.serviceType ) {
+				case GOOGLE_AD_CREDITS:
+					const currentVouchers = state[ siteId ]
+						? ( state[ siteId ][ GOOGLE_AD_CREDITS ] || [] )
+						: [];
+
+					const vouchers = currentVouchers.splice( 0 );
+					vouchers.push( action.voucher );
```

Comment:
Instead of adding more mutability in here, would you be interested in declaring it in one go?

```js
const vouchers = currentVouchers
	.splice( 0 )
	.concat( action.voucher );
```